scripts/asm-installer/install_asm

```diff
@@ -1411,9 +1424,25 @@ is_workload_identity_enabled() {
   if [[ "${ENABLED}" = 'null' ]]; then false; fi
 }
 
+get_hub_workload_identity() {
+  local IDENTITY_PROVIDER
+  local IDENTITY
+  IDENTITY_PROVIDER="$(kubectl get memberships membership -o=json | jq .spec.identity_provider)"
+  IDENTITY="$(echo "${IDENTITY_PROVIDER}" | sed 's/^\"https:\/\/gkehub.googleapis.com\/projects\/\(.*\)\/locations\/global\/memberships\/\(.*\)\"$/\1 \2/g')"
+  read -r ENVIRON_PROJECT_ID HUB_MEMBERSHIP_ID <<EOF
+${IDENTITY}
+EOF
+  MESHAPI_POST_DATA='{"workloadIdentityPools":["'${ENVIRON_PROJECT_ID}'.hub.id.goog"]}'
+  info "Cluster is in the Hub of project ${ENVIRON_PROJECT_ID} with Membership ID ${HUB_MEMBERSHIP_ID}"
+}
+
 enable_workload_identity(){
+  # Enable workload identity is not needed for Hub IDNS as the cluster needs to be registered with Hub IDNS before the installation script
+  if [[ "${CA}" == "mesh_ca" && "${ENABLE_HUB_IDNS}" -eq 1 ]]; then
```

Comment:
If Hub IDNS and Citadel isn't supported, let's check that combo in validate_args so we don't have to worry about it later.